tools/koch/koch.nim

```diff
@@ -134,15 +134,21 @@ proc getSourceMetadata(): tuple[hash, date: string] =
   var hash {.global.}: string
   var date {.global.}: string
 
-  if hash == "":
-    let hashCall = execCmdEx("git -C " & quoteShell(nimSource) & " rev-parse --verify HEAD")
-    if hashCall.exitCode == 0:
-      hash = hashCall.output.strip()
-
-  if date == "":
-    let dateCall = execCmdEx("git -C " & quoteShell(nimSource) & " log -1 --format=%cs HEAD")
-    if dateCall.exitCode == 0:
-      date = dateCall.output.strip()
+  if hash == "" or date == "":
+    try:
+      let releaseMetadata = parseFile(nimSource / "release.json")
+      hash = getStr releaseMetadata["commit"]
+      date = getStr releaseMetadata["commit_date"]
+    except OSError, IOError:
+      # If the file does not exist, then this is not a release tarball, try
+      # obtaining the data from git instead
+      let hashCall = execCmdEx("git -C " & quoteShell(nimSource) & " rev-parse --verify HEAD")
+      if hashCall.exitCode == 0:
+        hash = hashCall.output.strip()
+
+      let dateCall = execCmdEx("git -C " & quoteShell(nimSource) & " log -1 --format=%cs HEAD")
+      if dateCall.exitCode == 0:
+        date = dateCall.output.strip()
 
   result = (hash, date)
```

Comment:
so if these two aren't set for whatever reason we're still ok with that?

Reply:
Yea, other than missing information (that we only display in `--version`), I don't think there is much to worry about